pro-planner-client/src/redux/costSlice.js

```diff
@@ -1,69 +1,71 @@
+import { LOAD_STATUS } from "../constants";
+import axios from "axios";
+import { buildServerRoute } from "../helpers/Utils";
+
+export const getCostAsync = createAsyncThunk('cost/get', async({tripId}) => {
+    console.log('sending get request: ');
+    const response = await axios.get(buildServerRoute('cost', tripId));
+    return response.data;
+});
+
+export const addExpenseAsync = createAsyncThunk('cost/addExpense', async({tripId, userId, newItem, newItemAmount}) => {
+    const data = {
+        userName: userId,
+        itemName: newItem,
+        itemAmount: newItemAmount
+    }
+    const response = await axios.put(buildServerRoute('cost', 'addExpense', tripId), data);
+    return response.data;
+});
+
+export const removeExpenseAsync = createAsyncThunk('cost/removeExpense', async({tripId, userId, expenseId}) => {
+    const data = {
+        userName: userId,
+        expenseId: expenseId
+    }
+
+    const response = await axios.put(buildServerRoute('cost', 'removeExpense', tripId), data);
+    return response.data;
+});
 
 const costSlice = createSlice({
     name: 'cost',
     initialState: {
-        user1: {
-            userName: 'User 1',
-            expenses: {
-                expense1: {
-                    item: "Go Kart tickets",
-                    amount: 120
-                },
-                expense2: {
-                    item: "Movies",
-                    amount: 80
-                }
-            },
-        },
-        user2: {
-            userName: 'User 2',
-            expenses: {},
-        },
-        user3: {
-            userName: 'User 3',
-            expenses: {
-                expense1: {
-                    item: "PNE tickets",
-                    amount: 200
-                },
-                expense2: {
-                    item: "Coffee",
-                    amount: 10
-                }
-            },
-        },
-        user4: {
-            userName: 'User 4',
-            expenses: {
-                expense1: {
-                    item: "Parking",
-                    amount: 30
-                }
-            },
-        }
     },
     reducers: {
-        addExpense: (state, action) => {
-            const input = action.payload;
-            const newExpense = {
-                item: input.newItem,
-                amount: input.newItemAmount
-            }
-            state[input.userId].expenses[input.expenseId] = newExpense
-        },
-        removeExpense: (state, action) => {
-            const input = action.payload;
-            delete state[input.userId].expenses[input.expenseId];
-        },
-        addUser: (state, action) => {
-            const input = action.payload
-            const newUser = {
-                userName: input.name,
-                expenses: {}
-            }
-            state[input.id] = newUser
-        }
+    },
+    extraReducers: (builder) => {
+        builder.addCase(getCostAsync.pending, (state, action) => {
```

Comment:
Not sure if you are using all these states, but if you aren't you don't need to set all of these.